doc/user/content/sql/system-catalog/mz_internal.md

```diff
@@ -340,6 +340,25 @@ The view is defined as the transitive closure of [`mz_object_dependencies`](#mz_
 | `object_id`             | [`text`]     | The ID of the dependent object. Corresponds to [`mz_objects.id`](../mz_catalog/#mz_objects).                          |
 | `referenced_object_id`  | [`text`]     | The ID of the (possibly transitively) referenced object. Corresponds to [`mz_objects.id`](../mz_catalog/#mz_objects). |
 
+### `mz_optimizer_notices`
+
+The `mz_optimizer_notices` view contains a list of currently active optimizer
+notices emitted by the optimizer when planning materialized views and indexes.
+
+<!-- RELATION_SPEC mz_internal.mz_optimizer_notices -->
+| Field                   | Type                         | Meaning                                                                                                                                                                  |
+| ----------------------- | ---------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| `kind`                  | [`text`]                     | The notice kind. See https://materialize.com/docs/notices.                                                                                                               |
+| `message`               | [`text`]                     | A brief description of what concretely went wrong.                                                                                                                       |
+| `hint`                  | [`text`]                     | A high-level hint that tells the user what can be improved.                                                                                                              |
```

Comment:
Similarly, should this be `details`, for consistency with other objects? We use `hint` to refer to the actual concept of hints in [`mz_expected_group_size_advice`](https://materialize.com/docs/sql/system-catalog/mz_internal/#mz_expected_group_size_advice), and `details` to refer to additional helpful information (e.g. [`mz_sink_statuses`](https://materialize.com/docs/sql/system-catalog/mz_internal/#mz_sink_statuses)).

Reply:
Hmm, yes, good call about the naming clash with our other "hints" (which are given by users to the system). Maybe `suggestion`?

Reply:
@ggevay I think that @morsapaes means that a lot of other `mz_internal` tables have a `details` field (`mz_sink_statuses`, `mz_sink_status_history`, `mz_source_statuses`, `mz_source_status_history`).

@morsapaes The issue that we have here is that the `mz_optimizer_notices` structure is more detailed than these tables.

In order to properly render the notice in every possible form (in the UX, in `EXPLAIN` output and in `psql` notices)0 A, we need to differentiate between several fields:

- A title field is instance-agnostic and is determined by the `kind` (it will either be the `kind` or something looked up from an auxiliary `mz_optimizer_notice_kinds` table). This should be used when rendering notices in tabular form.
- A `message` field with a brief description of what concretely went wrong. Corresponds to the text of the `psql` notice and 
- A `hint` field that tells the user what can be improved. Corresponds to the text after the "HINT: " prefix in `psql` notices and in EXPLAIN output.
- An `action` field which if present will be a concrete incarnation of the remediation steps given in the above. This is because @sthm mentioned that it will be good if we can provide the concrete SQL syntax as opposed to aksing users to assemble it themselves. This is easier said than done in general, but for notices on simple DDL statements we should be able to do that (hence the `action_type` field).

Overall this table is internal and until somebody starts integrating this in the UX it won't be used at all. I suggest to have at least a version of the UX before we decide how and whether to rename the `hint` field because regardless of what we do there will be some inconsistency somewhere (either with other `mz_internal` tables or between representations of the same structure in the UX).

Reply:
We could also change the `hint:` prefix in `psql` notices to `suggestion:`. This would prevent somebody from confusing these "hints" (which we are giving to the user) from the "hints" that the user can give to the system in SQL. Or if you think this confusion is unlikely to actually arise then I'm ok with leaving it as "hint".

Reply:
> We could also change the hint: prefix in psql notices to suggestion:.

If @MaterializeInc/surfaces doesn't object I can do that (preferably in separate PR). I think the only reason against that is if somehow the "HINT" nomenclature is standard for the psql hints emitted by Postgres itself.

Reply:
Ok, if it's standard nomenclature in psql, then let's just leave it as it is.

Reply:
The tables I mentioned expose the `hint:` prefix in the `details` field, and AFAIU the idea is that other fields can be added in the future as needed (see #18009):

```sql
materialize=> SELECT * FROM mz_internal.mz_sink_status_history;
        occurred_at         | sink_id |  status  |            error            |                                                                                                       details
----------------------------+---------+----------+-----------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 2023-03-14 20:57:09.053+00 | u4      | starting |                             |
 2023-03-14 20:57:24.116+00 | u4      | starting |                             |
 2023-03-14 20:57:24.078+00 | u4      | stalled  | retriable transaction error | {"hint":"If you're running a single Kafka broker, ensure that the configs transaction.state.log.replication.factor, transaction.state.log.min.isr, and offsets.topic.replication.factor are set to 1 on the broker"}
(3 rows)
```

Not an opinion I hold strongly, though! Just double-checking how much we can standardize from the get go.